plugins/integrity_plugin.py

```diff
@@ -6,14 +6,16 @@
 
 
 class IntegrityChecker(Plugin):
-    def on_run_begin(self, pipeline: Pipeline) -> None:
+    def on_run_begin(self, pipeline: Pipeline) -> Pipeline:
         super().on_run_begin(pipeline)
```

Comment:
hmm could we in any way forcing people to call super.on_run_begin()? we'll probably never be able to verify if they did, and it's really easy to miss, as we don't require it anywhere else.
any logic we'd want to apply to all plugins, we could have in the runner, or somewhere rather!